src/main/java/org/spongepowered/api/item/recipe/crafting/ShapelessCraftingRecipe.java

```diff
@@ -0,0 +1,128 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.recipe.crafting;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.Lists;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * A ShapelessCraftingRecipe is a Recipe that does not have shape and just has a
+ * list of ingredients.
+ */
+public interface ShapelessCraftingRecipe extends CraftingRecipe {
+
+    /**
+     * Creates a new {@link Builder} to build a {@link ShapelessCraftingRecipe}.
+     *
+     * @return The new builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the ingredients for this shapeless recipe.
+     *
+     * @return The ingredients
+     */
+    ImmutableCollection<ItemStackSnapshot> getIngredients();
+
+    /**
+     * Represents a builder class to create {@link ShapelessCraftingRecipe}s.
+     *
+     * @see ShapelessCraftingRecipe
+     */
+    interface Builder extends ResettableBuilder<ShapelessCraftingRecipe, Builder> {
+
+        /**
+         * Sets the ingredients of this shapeless recipe.
+         * (Minecraft codes it so that it only takes one item of a type)
+         *
+         * @param ingredients The ingredients
+         * @return The builder
+         */
+        default Builder ingredients(ItemType... ingredients) {
+            return ingredients(Lists.newArrayList(ingredients).stream().map(
+                    type -> type.getTemplate().createStack()).collect(Collectors.toList()));
+        }
+
+        /**
+         * Sets the ingredients of this shapeless recipe.
+         * (Minecraft codes it so that it only takes one item of a type)
+         *
+         * @param ingredients The ingredients
+         * @return The builder
+         */
+        default Builder ingredients(ItemStack... ingredients) {
```

Comment:
In general, avoid only adding vararg methods like this, if possible, you should have single argument methods and then two argument methods with the second argument being a vararg. This would simplify some logic.